pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -489,9 +492,39 @@ func (c *Compactor) runCompact(ctx context.Context, logger log.Logger, job Job,
 			return level.Error(logger).Log("msg", "failed to compact new chunks", "err", err)
 		}
 
+		archivePath := filepath.Join(c.cfg.WorkingDirectory, storedBlock.BlockPath[strings.LastIndex(storedBlock.BlockPath, "/")+1:])
+		archiveFile, err := os.Create(archivePath)
+		if err != nil {
+			return err
+		}
+		defer archiveFile.Close()
+		defer func() {
+			err = os.Remove(archivePath)
+			if err != nil {
+				level.Error(logger).Log("msg", "removing archive file", "err", err, "file", archivePath)
+			}
+		}()
+		err = v1.TarGz(archiveFile, v1.NewDirectoryBlockReader(localDst))
+		if err != nil {
+			level.Error(logger).Log("msg", "creating bloom block archive file", "err", err)
+			return err
+		}
+		blockToUpload := bloomshipper.Block{}
+		blockToUpload.StartTimestamp = storedBlock.StartTimestamp
+		blockToUpload.EndTimestamp = storedBlock.EndTimestamp
+		blockToUpload.MinFingerprint = storedBlock.MinFingerprint
+		blockToUpload.MaxFingerprint = storedBlock.MaxFingerprint
+		blockToUpload.IndexPath = storedBlock.IndexPath
+		blockToUpload.Checksum = storedBlock.Checksum
+		blockToUpload.TenantID = storedBlock.TenantID
+		blockToUpload.TableName = storedBlock.TableName
```

Comment:
With exception of `IndexPath`, all fields are part of `Ref`. 
Just setting ref like on line 521 should be enough.

Reply:
even just setting `BlockRef` and `data`, essentially data is what has changed.